backend/manager/modules/vdsbroker/src/main/java/org/ovirt/engine/core/vdsbroker/builder/vminfo/LibvirtVmXmlBuilder.java

```diff
@@ -468,8 +470,12 @@ void writeCpu(boolean addVmNumaNodes) {
                 cores = vm.getCurrentCoresPerSocket();
                 threads = vm.getCurrentThreadsPerCore();
             } else {
-                int vcpus = VmInfoBuildUtils.maxNumberOfVcpus(vm);
-                sockets = vcpus / vm.getCpuPerSocket() / vm.getThreadsPerCpu();
+                if (vm.getCpuPinningPolicy() != CpuPinningPolicy.DEDICATED) {
+                    int vcpus = VmInfoBuildUtils.maxNumberOfVcpus(vm);
+                    sockets = vcpus / vm.getCpuPerSocket() / vm.getThreadsPerCpu();
+                } else {
+                    sockets = vm.getNumOfSockets();
```

Comment:
with the change in VmInfoBuildUtils.maxNumberOfVcpus, do we still need to differentiate here?

Reply:
yes

Reply:
why? wouldn't `VmInfoBuildUtils.maxNumberOfVcpus(vm) / vm.getCpuPerSocket() / vm.getThreadsPerCpu()` give us the same result?

Reply:
right i thought more about if we need to write here something else at all. yes, we can use now `maxNumberOfVcpus`.

Reply:
cool, the less logic we have here the better